### PR TITLE
[#2176] Move from Vue CLI to Vite

### LINKED ISSUE #2176: Replace Vue CLI with Vite
### What feature(s) would you like to see in RepoSense

Replace Vue CLI with Vite. There are [many benefits](https://vueschool.io/articles/vuejs-tutorials/how-to-migrate-from-vue-cli-to-vite/) to doing so.

### Is the feature request related to a problem?

- `vue/component-compiler-utils` (a dependency of Vue CLI) has a few security issues that have yet to be resolved and is unlikely to be resolved soon.
- Vue CLI itself is being put on maintenance mode.

### If possible, describe the solution
Install Vite and remove Vue CLI.

### Additional context
[Migration guide.](https://vueschool.io/articles/vuejs-tutorials/how-to-migrate-from-vue-cli-to-vite/)
#2143 for more information and things to take note of.

### PR DESCRIPTION
**Pull request by sopa301**
Fixes #2176

### Proposed commit message

```
Move from Vue CLI to Vite

Vue CLI has been put on maintenance mode and its dependencies have
security issues that have not been fixed for a while. Switching to Vite
also allows us to introduce Vitest for our unit testing.

Let's install Vite and remove Vue CLI.
```

### Other information

A few things worthy of discussion:
- The [Vite migration guide](https://vueschool.io/articles/vuejs-tutorials/how-to-migrate-from-vue-cli-to-vite/) suggests removing Babel (which supports older browsers), which has been done so. If we decide to retain support for them, we can do so with a [plugin](https://github.com/vitejs/vite/tree/main/packages/plugin-legacy).
- The Vite changes remove the ability to configure the title and source folder for the `html` file from the config file. However, now that the `html` file has been moved out of the `public` folder (by design of Vite), this may not be a big concern. We can use a `.env` file if we want to be able to edit the title and root folder from another file.
- The Cypress tests on CI did not work initially and had to be hotfixed by adding `"chromeWebSecurity": false` into the Cypress config file. Since this is not very extensible to other browsers, we should also find another way to get the tests to work.

### CODE REVIEW

**Review by asdfghjkxd — APPROVED** [review]
LGTM!

File: .gitignore
Comment (sopa301) [inline .gitignore:22]: New files generated when linting.

File: docs/_markbind/layouts/ug-sitenav.md
Comment (sopa301) [inline docs/_markbind/layouts/ug-sitenav.md:10]: Fixes required by linter.

File: frontend/.eslintrc.json
Comment (sopa301) [inline frontend/.eslintrc.json:3]: Done as part of migration guide.

File: frontend/src/main.ts
Comment (sopa301) [inline frontend/src/main.ts:7]: Required as the move to Vite somehow broke the existing highlighting mechanism.

File: frontend/src/styles/highlight-js-style.css
Comment (sopa301) [inline frontend/src/styles/highlight-js-style.css:1]: File added to linting as it was imported into `main.ts`

**Review by MarcusTXK — APPROVED** [review]
LGTM, this seems to work locally, and all test cases are passing. 

Regarding browser support, it might be good if we could document it somewhere that we support most mobile and desktop evergreen browsers, as they [have near full support for all ES6 features](https://compat-table.github.io/compat-table/es6/) (Mainly just looks like Internet Explorer 11 that isn't supported). Perhaps we can add this in either the User Guide Appendix: Troubleshooting or a little footnote somewhere in the Developer guide for easy reference in the future.

File: frontend/src/main.ts
Comment (ckcherry23) [reply frontend/src/main.ts:7]: Seems like the code syntax highlighting is working for me even without this import statement. Could you please check again?

**Review by ckcherry23 — CHANGES_REQUESTED** [review]

File: frontend/package.json
Comment (sopa301) [reply frontend/package.json]: Okay, I've updated the command!

**Review by ckcherry23 — APPROVED** [review]
LGTM, thanks @sopa301 for suggesting and handling this migration to Vite! It's a great move that should boost performance and streamline our configuration. 

Since it's a significant architectural shift, let's try to get one more approval before merging.

**Review by vvidday — APPROVED** [review]
LGTM, great job!

**Pull request merged**
